cli/Sources/TuistHasher/TargetScriptsContentHasher.swift

```diff
@@ -64,4 +84,63 @@ public struct TargetScriptsContentHasher: TargetScriptsContentHashing {
         }
         return try contentHasher.hash(stringsToHash)
     }
+
+    // MARK: - Private
+
+    private func processInputFileListPath(
+        _ fileListPath: AbsolutePath,
+        sourceRootPath: AbsolutePath,
+        pathsToHash: inout [AbsolutePath],
+        stringsToHash: inout [String]
+    ) async {
+        if fileListPath.pathString.contains("$") {
+            stringsToHash.append(fileListPath.relative(to: sourceRootPath).pathString)
+            Logger.current.notice(
+                "The path of the file \'\(fileListPath.url.lastPathComponent)\' is hashed, not the content. Because it has a build variable."
+            )
+        } else if fileListPath.extension == "xcfilelist" {
+            await processXCFileList(
+                fileListPath,
+                sourceRootPath: sourceRootPath,
+                pathsToHash: &pathsToHash,
+                stringsToHash: &stringsToHash
+            )
+        } else {
+            pathsToHash.append(fileListPath)
+        }
+    }
+
+    private func processXCFileList(
```

Comment:
would call this `hashXCFileList` and return the hashed value instead of using `inout` here, too. Hashing strings is a fast operation.